### DISCUSSION TITLE
Connecting Open WebUI to Langchain pipeline and RLHF evaluation to Langfuse

### DISCUSSION THREAD

**Discussion by eresque** [open-webui/open-webui#2652] [Q&A]
I have a local Langchain RAG system, with my own hosted vector database and retriever. I wanted to use Open WebUI as an interface, because it has many useful features, however, I've encountered some problems  and would appreciate your advice!
1. The main issue is that I've only managed to  connect Open WebUI  directly to my llama.cpp server, however for a fully functional RAG system I need to be able to take an input query from Open WebUI and send it to my uvicorn server, which will then feed it to the langchain pipeline and produce the output. Is there any way to do it?
2. Also I'm only  being able to get RLHF evaluations via Export chats option in settings. Is there any way to send them to Langfuse directly, instead of parsing exported json files?

**Comment by tjbck** [comment]
Related: https://github.com/open-webui/pipelines

**Reply by ther3zz** [reply]
> Related: https://github.com/open-webui/pipelines


@tjbck Would it be possible to include the feedback received from a user in the example langfuse pipline (https://github.com/open-webui/pipelines/blob/main/examples/filters/langfuse_filter_pipeline.py)?

Seems that feedback is not currently sent to langfuse at the moment (or am I just blind 😆)

**Comment by pedroallenrevez** [comment]
Also wondering about this, how can one set up a RLHF connection straight to langfuse?

**Comment by bartonzzx** [comment]
Maybe using OpenWebui Pipelines can help you, and I wrote an example of this function.Check it out: https://github.com/open-webui/pipelines/tree/main/examples/pipelines/integrations/langgraph_pipeline

**Comment by sergeyk-ytree** [comment]
Also interested in sending evaluations to Langfuse.
The only way I found for now is patching a router evaluations.py to send feedback data. 
[evaluations.py](https://github.com/user-attachments/files/24171398/evaluations.py)
Also to attach evaluations to correct traces I modified the langfuse integration pipeline to use deterministic trace id:
`trace_id = self.langfuse.create_trace_id(seed=chat_id)`
Are there other ways to do that?
